src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```
@@ -765,27 +766,39 @@ private void requestSimpleUpload(Intent data, int resultCode) {
-//                    remotePath += displayName + DisplayUtils.getComposedFileExtension(filePath);
-//
-//                }
-//                // and what happens in case of error?; wrong target name for the upload
-//            } finally {
-//                cursor.close();
-//            }
-            // Pending to be fixed
-            Toast.makeText(this, R.string.common_error_unknown, Toast.LENGTH_SHORT).show();
-            return;
+        if (selectedFileUri.toString().startsWith(UriUtils.URI_CONTENT_SCHEME)) {
+            Cursor cursor = getContentResolver().query(selectedFileUri, null, null, null, null);
+            try {
+                if (cursor != null && cursor.moveToFirst()) {
+                    String displayName = cursor.getString(cursor.getColumnIndex(
+                            OpenableColumns.DISPLAY_NAME));
+                    Log_OC.v(TAG, "Display Name: " + displayName);
+
+                    displayName = displayName.replace(File.separatorChar, '_');
+
+                    remotePath += displayName;
+
+                    // Check if extension is not included in file display name and add file extension
+                    int pos = displayName.lastIndexOf('.');
+                    if (pos < 0) {
+                        remotePath += DisplayUtils.getFileExtension(filePath);
+                    }
 
+                    // URi and remote path parameters
+                    Uri[] uris = new Uri[]{selectedFileUri};
+                    String[] remotePaths = new String[]{remotePath};;
+
+                    // Call to copy and then upload the selected file
+                    copyThenUpload(uris, remotePaths);
+
+                    cursor.close();
+                    return;
+
+                }
+                // and what happens in case of error?; wrong target name for the upload
+            } catch (Exception e) {
+                Log_OC.e(TAG, "Error while trying to copy and upload a schema type content file ", e);
```

This will end in a silent error; we should show the ```Toast``` that was introduced in version 2.0.0, previous line number 786